### ISSUE TITLE
[navigator] Sometimes the tree does not update when creating a new file

### ISSUE THREAD

**Issue by geropl** [theia-ide/theia#1354]
_Sometimes_ (read: **once**) the navigator tree does not show the newly created and saved file (with right on the navigator folder -> create file).

Needed a "collapse" and "expand" to suddenly pop up.

**Comment by epatpol** [comment]
@akosyakov do you think it could be related to the recent fs-watcher events changes #1312 ?

**Comment by svenefftinge** [comment]
I assume you checked you didn't run out of file handles so the workspace was actually watched?

**Comment by geropl** [comment]
@svenefftinge Yes. The logs did not contain any errors that were (obviously) related to the file watcher. 

> [2018-02-22T16:08:02.625Z] INFO: [...] WARNING: Promise with no error callback:38 []
[2018-02-22T16:08:02.626Z] INFO: [...] [object Object] []
[2018-02-22T16:08:02.627Z] INFO: [...] WARNING: Promise with no error callback:39 []
[2018-02-22T16:08:02.627Z] INFO: [...] [object Object] []

But maybe the errors mentioned above are thrown in the watcher?

**Comment by tsmaeder** [comment]
I believe this is an instance of https://github.com/Axosoft/nsfw/issues/60 and as such a dup of https://github.com/eclipse-theia/theia/issues/5738

**Comment by sgraband** [comment]
File watcher subsystem completely reworked; nsfw replaced with `@parcel/watcher`
